python/src/aiconfig/editor/client/src/components/prompt/PromptContainer.tsx

```diff
@@ -158,54 +175,65 @@ export default memo(function PromptContainer({
   const inputSchema = promptSchema?.input;
 
   return (
-    <Flex justify="space-between" w="100%">
-      <Card
-        withBorder
-        className={`${classes.cellStyle} cellStyle`}
-        ref={cellInputOutputRef}
-      >
-        <Flex direction="column">
-          <Flex justify="space-between" mb="0.5em">
-            <PromptName
-              promptId={promptId}
-              name={prompt.name}
-              onUpdate={onChangeName}
-            />
-            <ModelSelector
-              getModels={getModels}
-              prompt={prompt}
-              onSetModel={updateModel}
-              defaultConfigModelName={defaultConfigModelName}
-            />
-          </Flex>
-          <PromptInputRenderer
-            input={prompt.input}
-            schema={inputSchema}
-            onChangeInput={onChangeInput}
-            onCancelRun={onCancelRun}
-            onRunPrompt={runPrompt}
-            isRunning={prompt._ui.isRunning}
-            isRunButtonDisabled={isRunButtonDisabled}
-          />
-
-          {prompt.outputs && prompt.outputs.length > 0 && (
-            <>
-              <PromptOutputBar />
-              <PromptOutputsRenderer outputs={prompt.outputs} />
-            </>
-          )}
-        </Flex>
-      </Card>
-      <div className={`${classes.sidePanel} sidePanel`}>
-        <PromptActionBar
-          defaultConfigModelName={defaultConfigModelName}
-          prompt={prompt}
-          promptSchema={promptSchema}
-          onUpdateModelSettings={updateModelSettings}
-          onUpdateParameters={updateParameters}
-          onUpdatePromptMetadata={updatePromptMetadata}
+    <>
+      {!readOnly && (
+        <PromptMenuButton
+          showDeleteOutput={!!prompt.outputs?.length}
```

Comment:
nit: feel `prompt.outputs?.length > 0` is easier to read and I don't really like JS implicitly "falsey-ness" but don't really care

Reply:
it actually complains if you use this syntax unelss you do a `(prompt.outputs?.length || 0) > 0` it seems, so thats why i kept it with this way